src/app/loadout/plug-drawer/PlugDrawer.tsx

```diff
@@ -82,27 +92,34 @@ export default function PlugDrawer({
   }, [isPhonePortrait, filterInput]);
 
   const onPlugSelected = useCallback(
-    (mod: PluggableInventoryItemDefinition) => {
+    (plugSetHash: number, mod: PluggableInventoryItemDefinition) => {
+      // TODO (ryan) use immer
       setSelected((oldState) => {
-        const newState = [...oldState];
-        newState.push(mod);
+        const newState = { ...oldState };
+        const newSelected = newState[plugSetHash] || [];
+        newSelected.push(mod);
```

Comment:
Nope, you're still mutating `oldState` directly - `{ ...oldState }` is a shallow copy.

Reply:
Ah thanks, I won't do it in this PR but I am going to come back and replace this with immer